v2/anomalydetection/backend-setup.mdx

```diff
@@ -0,0 +1,683 @@
+    phoneNumber?: string,
+    password?: string,
+    requestId?: string,
+    bruteForce?: {
+        keyStr: string,
+        maxRequests: {
+            limit: number,
+            perTimeIntervalMS: number
+        }[]
+    }[]
+}): Promise<{
+    status: "GENERAL_ERROR",
+    message: string
+} | undefined> {
+    let requestBody: {
+        email?: string;
+        phoneNumber?: string;
+        actionType?: string;
+        requestId?: string;
+        passwordHash?: string;
+        bruteForce?: {
+            keyStr: string;
+            maxRequests: {
+                limit: number;
+                perTimeIntervalMS: number;
+            }[];
+        }[];
+    } = {}
+
+    if (input.requestId !== undefined) {
+        requestBody.requestId = input.requestId;
+    }
+
+    if (input.password !== undefined) {
+        let shasum = createHash('sha1');
+        shasum.update(input.password);
+        const passwordHash = shasum.digest('hex');
+        requestBody.passwordHash = passwordHash;
+    }
+    requestBody.bruteForce = input.bruteForce;
+    requestBody.email = input.email;
+    requestBody.phoneNumber = input.phoneNumber;
+    requestBody.actionType = input.actionType;
+
+    let response = await axios.post(ANOMALY_DETECTION_API_URL, requestBody);
+    let responseData = response.data;
+
+    if (!responseData.bruteForce.result) {
+        throw new Error("Too many requests. Please try again later.");
+    }
+
+    if (responseData.isBreachedPassword) {
+        // ... your code here ...
+    }
+
+    if(responseData.isNewDevice) {
+        // ... your code here ...
+    }
+
+    if(responseData.isImpossibleTravel) {
+        // ... your code here ...
+    }
+
+    if(responseData.numberOfUniqueDevicesForUser) {
+        // ... your code here ...
+    }
+    
+    if(responseData.requestIdInfo) {
+        // handle stuff like bot detection, vpn detection, tor detection, etc
+        // ... your code here ...
+    }
+
+    return responseData;
+}
+
+const getBruteForce = (email: string) => [{
+    keyStr: `${email}`,
+    maxRequests: [
+        {
+            limit: 1,
+            perTimeIntervalMS: 1000,
+        },
+        {
+            limit: 100,
+            perTimeIntervalMS: 60 * 1000 * 60,
+        }
+    ]
+}];
+
+
+// backend
+SuperTokens.init({
+    appInfo: {
+        apiDomain: "...",
+        appName: "...",
+        websiteDomain: "..."
+    },
+    supertokens: {
+        connectionURI: "...",
+    },
+    recipeList: [
+        EmailPassword.init({
+            // highlight-start
+            override: {
+                functions: (originalImplementation: APIInterface): APIInterface => {
+                    return {
+                        ...originalImplementation,
+                        signUpPOST: async function (input: any) {
+                            // There is no need to generate a request ID when signing up
+
+                            const bruteForce = getBruteForce(input.email);
```

Comment:
better naming  that bruteForce. Make something like bruteForceConfig. Also, the key has to have `signup` in it so attempts for sign up, sign in do not use the same key..

Reply:
See https://github.com/rishabhpoddar/kubra/blob/main/backend/config.ts#L163